Refactor chat_logging.py and config.py

gracebrazil28 opened the pull request:
- **chat_logging.py**:
  - Moved chat logging functions to a separate module for better organization.
  - Added new logging functions to handle different types of chat reactions.

- **config.py**:
  - Centralized configuration settings
  - Introduced a Config class to handle environment variables and settings.
  - Updated the application to use the new configuration class.

cs1090218 commented on config.py:
```diff
@@ -0,0 +1,56 @@
+#####################################################################################################################
+import os
+
+class Config:
```
I was envisioning the following, let me know what you think:
1- rename the file to constants.py
2- Have all the constants set here in this file directly (don't really need class to encapsulate them)
3- Move URL_TOKEN, ACCESS_TOKEN, USER_NAME, USERNAME and USER_EMAIL out of this file, since they aren't constants and will be set by the app after making some API calls
4- Use these vars in the main.py wherever required directly like constants.IS_DEVELOPER_VIEW, constants.MCM_URL, etc. This way we can remove the lines like "IS_DEVELOPER_VIEW = Config.IS_DEVELOPER_VIEW" at the top of main.py and make the file more concise and also looking at var named constants.MCM_URL, it is known that these are constants in another file.

Let me know what you think.

gracebrazil28 replied:
Thanks for the feedback
1- Yes, that sounds more intuitive since we want the constants here
2- The idea for setting them as a class was because it prevents it from being accidentally modified, modifying them needs to be through methods which we can easily verify. If this seems too much we can do away as long as we're careful with it.
3- Would these remain global variables? Sounds good to me.
4- Yeah it makes it easier,  Config.IS_DEVELOPER_VIEW was too clunky when I was setting them.

cs1090218 replied:
2- Unfortunately it is still possible to modify the variables even when they are part of class.
class A:
    class_var1 = 1

A.class_var1 = 3
print (A.class_var1)  # prints 3

3- These will be global vars but not constants so they will follow "var_name" format as opposed to "CONST_NAME" format.

gracebrazil28 commented:
@cs1090218 
I've changed the following for the following comments:
1- rename the file to constants.py
4- Use these vars in the main.py wherever required directly

2- Have all the constants set here in this file directly (don't really need class to encapsulate them)
--> For 2, the rest don't need an encapsulating class but for username and access token, I still put them in a config class (basically variables that will be updated after login or get_access_token function call).  With a class, configuration settings are set/managed in one place and is the one source of truth. When a value in the Config class is updated, it is immediately available to all parts, **not just main,**  that access this class. 
Without it, I basically had to leave reaction handling in main.py to make it work since it needed an updated USERNAME and ACCESS_TOKEN. However, it doesn't make sense since the function belonged in the chat_logging.py module.  

3- Move MCM_URL, URL_TOKEN, ACCESS_TOKEN, USER_NAME, USERNAME and USER_EMAIL out of this file, since they aren't constants and will be set by the app after making some API calls
--> I took out the ones you've mentioned except for the three that will be accessed in the chat_logging module and updated in main.py: ACCESS_TOKEN, USER_NAME, USERNAME.

I tested these changes already, feel free to do some additional testing if needed.

cs1090218 commented on constants.py:
```diff
@@ -0,0 +1,61 @@
+GET_ACCESS_TOKEN_URL = COGNITO_DOMAIN + "/oauth2/token"
+GET_USER_INFO_URL = COGNITO_DOMAIN + "/oauth2/userInfo"
+
+class Config:
```
We can rename this to UserConfig since this only has user data.

cs1090218 replied:
We can pull this out into another file actually. user_data.py or something as it doesn't make sense to have non-constants in constants file.

cs1090218 commented on main.py:
```diff
@@ -101,21 +65,21 @@ def get_user_info_header(access_token):
-        global ACCESS_TOKEN
-        ACCESS_TOKEN = access_token
+        # Update Access token in constants/config file 
+        Config.ACCESS_TOKEN = access_token 
```
Config isn't imported in this file.

cs1090218 commented on constants.py:
```diff
@@ -0,0 +1,50 @@
+#####################################################################################################################
+# Description:
+# The constants.py module defines the configuration settings and constants.
+# Class UserConfig is used so that other modules are able to 
```
You can remove this docstring now since this class is not in this file anymore.

cs1090218 reviewed (approved):
LGTM
